b2luigi/batch/processes/gbasf2.py

```diff
@@ -0,0 +1,764 @@
+            gbasf2_command_str += " --force "
+
+        # estimated cpu time per sub-job in minutes
+        cpu_minutes = get_setting("gbasf2_cputime", default=False, task=self.task)
+        if cpu_minutes is not False:
+            gbasf2_command_str += f" --cputime {cpu_minutes} "
+
+        # estimated number or processed events per second
+        evtpersec = get_setting("gbasf2_evtpersec", default=False, task=self.task)
+        if evtpersec is not False:
+            gbasf2_command_str += f" --evtpersec {evtpersec} "
+
+        # gbasf2 job priority
+        priority = get_setting("gbasf2_priority", default=False, task=self.task)
+        if priority is not False:
+            assert 0 <= priority <= 10, "Priority should be integer between 0 and 10."
+            gbasf2_command_str += f" --priority {priority} "
+
+        # gbasf2 job type (e.g. User, Production, ...)
+        jobtype = get_setting("gbasf2_jobtype", default=False, task=self.task)
+        if jobtype is not False:
+            gbasf2_command_str += f" --jobtype {jobtype} "
+
+        # additional basf2 options to use on grid
+        basf2opt = get_setting("gbasf2_basf2opt", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" --basf2opt='{basf2opt}' "
+
+        # optional string of additional parameters to append to gbasf2 command
+        gbasf2_additional_params = get_setting("gbasf2_additional_params", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" {gbasf2_additional_params} "
+
+        gbasf2_command = shlex.split(gbasf2_command_str)
+        return gbasf2_command
+
+    def _write_path_to_file(self):
+        """
+        Serialize and save the ``basf2.Path`` returned by ``self.task.create_path()`` to a python pickle file.
+        """
+        try:
+            path = self.task.create_path()
+        except AttributeError as err:
+            raise Exception(
+                "Gbasf2 batch process can only used with tasks that generate basf2 paths with "
+                "a ``create_path()`` method, e.g. are an instance of ``Basf2PathTask``."
+            ) from err
+
+        path.add_module("Progress")
+        b2pp.write_path_to_file(path, self.pickle_file_path)
+
+    def _create_wrapper_steering_file(self):
+        """
+        Create a steering file to send to the grid that executes the pickled
+        basf2 path from ``self.task.create_path()``.
+        """
+        # read a jinja2 template for the steerinfile that should execute the pickled path
+        template_file_path = os.path.join(self._file_dir, "templates/gbasf2_steering_file_wrapper.jinja2")
+        with open(template_file_path, "r") as template_file:
+            template = Template(template_file.read())
+            # replace some variable values in the template
+            steering_file_stream = template.stream(
+                pickle_file_path=self.pickle_file_path,
```

Comment:
Is this path also available on the grid? I mean you copy the file on the submit call, but if I understand correctly you replace the template string with the full path here, or?

Reply:
I use absolute paths, but gbasf2/dirac recreates the directory structure of the path in which gbasf2 is called on the grid inside the job input sandbox and probably chroot's into it or something like that.  So far it seemed to work for my use-cases. Relative paths would probably also have worked.